src/Relude/List/NonEmpty.hs

```diff
@@ -9,21 +15,76 @@ Maintainer:  Kowainik <xrom.xkov@gmail.com>
 Stability:   Stable
 Portability: Portable
 
-This module contains safe functions to work with list type in terms of 'NonEmpty'.
+This module contains reexports from "data.List.NonEmpty" and safe functions to
+work with list type in terms of 'NonEmpty'.
+
+Note, that Relude reexports 'head', 'tail', 'init', 'last' from
+"Data.List.NonEmpty" instead of the "Data.List", so these functions are safe to
+use.
+
+@relude@ also provides custom type error for better experience with transition
+from lists to 'NonEmpty' with those functions.
+
+
+Let's examine the behaviour of the @relude@ list functions comparing to the
+corresponding @base@ one on the example of the 'head' function:
+
++-----------------+-----------------------------------------+
+|                 |         'head'                          |
++=================+=========================================+
+| __@base@__      | @[a] -> a@                              |
++-----------------+-----------------------------------------+
+| __@relude@__    | @'NonEmpty' a -> a@                     |
++-----------------+-----------------------------------------+
+| Example         | @> 'Data.List.head' [1..5]@             |
+| with list       +-----------------------------------------+
+| __@base@__      | @1@                                     |
++-----------------+-----------------------------------------+
+| Example         | @> 'Data.List.head' []@                 |
+| with empty list +-----------------------------------------+
+| __@base@__      |@*** Exception: Prelude.head: empty list@|
++-----------------+-----------------------------------------+
+| Example         |@> 'head' $ 1 :| [2..5]@                 |
+| with 'NonEmpty' +-----------------------------------------+
+| __@relude@__    | @1@                                     |
++-----------------+-----------------------------------------+
+| Example         | @> 'viaNonEmpty' 'head' [1..5]@         |
+| with list       +-----------------------------------------+
+| __@relude@__    | @'Just' 1@                              |
```

Comment:
This table is just so awesome! :heart: :sweat: 

I was also thinking about the table with the difference in type signatures, like:

```
      base           relude
head  [a] -> a    NonEmpty a -> a
tail  [a] -> [a]   NonEmpty a -> [a]
last ...
init ...
```

What do you think? Would it make sense to have such table here or in the `Relude.List` module to have more modules with tables? I just though that it could be useful to see the difference of list functions in terms of type signatures, to quickly get the idea...